### PR TITLE
[AC-5919] Add Clearance-based permission checks for allocator API calls

### PR DESCRIPTION
**Pull request by jonkiparsky**
* added mechanism for Clearance-based permission checks on impact-api
** note that this mechanism duplicates some code from accelerate, and uses a slightly different approach. I did try to hoist up the ClearanceManager and some code from permission_checks to django-accelerator, so it could be shared between accelerate and impact, but this failed due to cross-repo model sharing issues. 
* added Global Operations Manager permission checks on analyze_judging_round and clone_criteria API calls
* altered test settings to squelch logging output (otherwise, it echoed with test output)
** oops, this doesn't seem to have worked after all. attempting to fix
** set logging level to "info" and created a ticket about silent logging for tests
* If this work is acceptable, will update AC-5924 with the details of these changes to preserve memory

To test: 
* verify that the added tests are reasonable
* replicate the added tests by hitting these endpoints with users holding just v1 permissions, and with users holding global ops manager clearance. Should fail and succeed, respectively.

### CODE REVIEW

**Comment by pbx** [comment]
I have a review in progress but am pausing it while i work on urgent ticket [AC-5970](https://masschallenge.atlassian.net/browse/AC-5970)

**Review by pbx — CHANGES_REQUESTED** [review]
(Removed a note I had written about this not working -- user error! I had failed to add my user to `v1_clients` so I was getting only 403s)

**Review by pbx — APPROVED** [review]

**Review by pbx — CHANGES_REQUESTED** [review]
I resolved my previous "request changes" review when I fixed my functional problem, then remembered a couple minor things I noted in it that still could use attention. So, back in "request changes" for the moment.

File: web/impact/impact/permissions.py [hunk 1]
```diff
@@ -26,6 +32,64 @@
     "DELETE": "delete",
 }
 
+# NOTE: following definitions are duplicated from mc.ClearanceManager and
+# mc.permission_checks
+# TO DO: resolve this duplication
```
[the remark this answers is not in the document]
Comment (jonkiparsky) [reply web/impact/impact/permissions.py:37]: Won't fix. Any useful hint would require tracing back through a bunch of things I tried two weeks ago, which are no longer in my head. I'll assume that the comment is enough indication that some attempt has been made and has failed. Hopefully that will warn off the optimists.

File: web/impact/impact/tests/test_analyze_judging_round_view.py [hunk 2]
```diff
@@ -8,29 +8,43 @@
...
 from accelerator.tests.contexts import AnalyzeJudgingContext
+
 from impact.tests.api_test_case import APITestCase
 from impact.tests.utils import assert_fields
 from impact.v1.views import AnalyzeJudgingRoundView
 
 
 class TestAnalyzeJudgingRoundView(APITestCase):
-    def test_get(self):
+    def test_global_operations_permission_required(self):
```
[the remark this answers is not in the document]
Comment (jonkiparsky) [reply web/impact/impact/tests/test_analyze_judging_round_view.py]: renamed this test.

File: web/impact/impact/permissions.py [hunk 3]
```diff
@@ -14,6 +15,11 @@
 
 from impact.utils import model_name_case
 
+from accelerator_abstract.models.base_clearance import (
```
[the remark this answers is not in the document]
Comment (jonkiparsky) [reply web/impact/impact/permissions.py]: done

**Review by pbx — APPROVED** [review]

**Pull request merged**